pkg/integrations/v2/app_o11y_receiver/app_o11y_receiver.go

```diff
@@ -88,27 +90,45 @@ func (c *Config) NewIntegration(l log.Logger, globals integrations.Globals) (int
 	sourcemapLogger := log.With(l, "subcomponent", "sourcemaps")
 	sourcemapStore := sourcemaps.NewSourceMapStore(sourcemapLogger, c.ExporterConfig.SourceMaps, reg, nil, nil)
 
-	logsInstance := globals.Logs.Instance(c.ExporterConfig.LogsInstance)
-	lokiExporter := exporters.NewLokiExporter(
-		l,
-		exporters.LokiExporterConfig{
-			LokiInstance:     logsInstance,
-			Labels:           c.ExporterConfig.LogsLabels,
-			SendEntryTimeout: c.ExporterConfig.LogsSendTimeout,
-		},
-		sourcemapStore,
-	)
-
 	receiverMetricsExporter := exporters.NewReceiverMetricsExporter(exporters.ReceiverMetricsExporterConfig{
 		Reg: reg,
 	})
 
 	var exp = []exporters.AppO11yReceiverExporter{
-		// Loki
-		lokiExporter,
 		receiverMetricsExporter,
 	}
 
+	if len(c.ExporterConfig.LogsInstance) > 0 {
+		logsInstance := globals.Logs.Instance(c.ExporterConfig.LogsInstance)
+		if logsInstance == nil {
+			return nil, fmt.Errorf("logs instance \"%s\" not found", c.ExporterConfig.LogsInstance)
+		}
+		lokiExporter := exporters.NewLokiExporter(
+			l,
+			exporters.LokiExporterConfig{
+				LokiInstance:     logsInstance,
+				Labels:           c.ExporterConfig.LogsLabels,
+				SendEntryTimeout: c.ExporterConfig.LogsSendTimeout,
+			},
+			sourcemapStore,
+		)
+		exp = append(exp, lokiExporter)
+	}
+
+	if len(c.ExporterConfig.TracesInstance) > 0 {
+		tracesInstance := globals.Tracing.Instance(c.ExporterConfig.TracesInstance)
+		if tracesInstance == nil {
+			return nil, fmt.Errorf("traces instance \"%s\" not found", c.ExporterConfig.TracesInstance)
+		}
+
+		factory := tracesInstance.GetFactory(component.KindReceiver, "push_receiver")
```

Comment:
Nit: use push_receiver's const

```suggestion
		factory := tracesInstance.GetFactory(component.KindReceiver, pushreceiver.TypeStr)
```